src/system_info/system_info_network_ivi.cc

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "system_info/system_info_network.h"
+
+SysInfoNetwork::SysInfoNetwork()
+    : type_(SYSTEM_INFO_NETWORK_UNKNOWN) {
+  PlatformInitialize();
+}
+
+void SysInfoNetwork::PlatformInitialize() {
+  conn_ = system_info::GetDbusConnection();
+  if (!conn_)
+    return;
+  modem_path_ = system_info::OfonoGetModemPath(conn_);
+}
+
+SysInfoNetwork::~SysInfoNetwork() {
+  if (!conn_)
+    return;
+  g_dbus_connection_signal_unsubscribe(conn_,
+      cellular_network_type_changed_watch_);
+  g_dbus_connection_close_sync(conn_, NULL, NULL);
+  conn_ = NULL;
+}
+
+void SysInfoNetwork::StartListening() {
+  vconf_notify_key_changed(VCONFKEY_NETWORK_STATUS,
+      (vconf_callback_fn)OnNetworkTypeChanged, this);
+
+  if (!conn_ || !modem_path_)
+    return;
```

Comment:
This checking is for the following <code>g_dbus*</code> call not related with <code>vconf</code>. So put it here is correct :)